pkg/cache/sql/informer/listoption_indexer.go

```diff
@@ -176,7 +214,24 @@ func (l *ListOptionIndexer) afterUpsert(key string, obj any, tx db.TXClient) err
 	return nil
 }
 
-func (l *ListOptionIndexer) afterDelete(key string, tx db.TXClient) error {
+// labels are stored in tables that shadow the underlying object table for each GVK
+func (l *ListOptionIndexer) addLabels(key string, obj any, tx db.TXClient) error {
+	k8sObj, ok := obj.(*unstructured.Unstructured)
+	if !ok {
+		logrus.Debugf("UpsertLabels: Error?: Can't convert obj into an unstructured thing.")
+		return nil
+	}
+	incomingLabels := k8sObj.GetLabels()
+	for k, v := range incomingLabels {
+		err := tx.StmtExec(tx.Stmt(l.upsertLabelsStmt), key, k, v)
+		if err != nil {
+			return err
```

Comment:
Straightforward to do, but I don't see a straightforward way to do unit tests with that change because it's run from an `afterUpsert` handler in a different module.

Reply:
I do not see how changing the returned error type should make testing any harder - if anything, in my mind it should make it easier to detect.

What am I missing?

Reply:
I guess I could test the `after-upsert` handler directly. But I'm going to leave this, because better testing would involve an actual database, and then triggering addition of an object and verifying the expected labels have been added after. So no more work to be done here.